### ISSUE TITLE
Doesn't work with Spring Boot

### ISSUE THREAD

**Issue by rios0rios0** [javabeanz/owasp-security-logging#60]
Hello guys!
I don't know why but with the Spring Boot, nothing of the Markers works...
My ```pom.xml```:
```
<dependency>
    <groupId>org.springframework</groupId>
    <artifactId>spring-context-support</artifactId>
    <version>5.3.4</version>
</dependency>
<dependency>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-web</artifactId>
    <exclusions>
        <exclusion>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-logging</artifactId>
        </exclusion>
    </exclusions>
</dependency>
<dependency>
    <groupId>org.projectlombok</groupId>
    <artifactId>lombok</artifactId> <!-- annotations to simplify the code -->
    <version>1.18.18</version>
</dependency>
<dependency>
    <groupId>org.owasp</groupId>
    <artifactId>security-logging-log4j</artifactId>
    <version>LATEST</version>
</dependency>
```
My code:
```
public final class RequestLogger extends Logger {

  private static final org.slf4j.Logger log = LoggerFactory.getLogger(RequestLogger.class);

  public RequestLogger() {
    super();
    // log shell environment variables
    log.info(SecurityMarkers.CONFIDENTIAL, "password={}", "aa");
  }
  // simplified to read easier
}
```
Output:
```
2021-06-28 19:33:09,201 restartedMain WARN Log4j does not support detached Markers. Returned Marker [RESTRICTED] will be unchanged.
2021-06-28 19:33:09,205 restartedMain WARN Log4j does not support detached Markers. Returned Marker [CONFIDENTIAL] will be unchanged.
2021-06-28 19:33:09,209 restartedMain WARN Log4j does not support detached Markers. Returned Marker [SECRET] will be unchanged.
2021-06-28 19:33:09,212 restartedMain WARN Log4j does not support detached Markers. Returned Marker [TOPSECRET] will be unchanged.
2021-06-28 19:33:09,217 restartedMain WARN Log4j does not support detached Markers. Returned Marker [SECURITY SUCCESS] will be unchanged.
2021-06-28 19:33:09,221 restartedMain WARN Log4j does not support detached Markers. Returned Marker [SECURITY FAILURE] will be unchanged.
2021-06-28 19:33:09,225 restartedMain WARN Log4j does not support detached Markers. Returned Marker [SECURITY AUDIT] will be unchanged.
2021-06-28 19:33:09,228 restartedMain WARN Log4j does not support detached Markers. Returned Marker [EVENT SUCCESS] will be unchanged.
2021-06-28 19:33:09,232 restartedMain WARN Log4j does not support detached Markers. Returned Marker [EVENT FAILURE] will be unchanged.
2021-06-28 19:33:09.236  INFO 82744 --- [  restartedMain] c.i.i.a.l.RequestLogger                  : password=aa
```
I already read this: https://github.com/javabeanz/owasp-security-logging/issues/26
Can you help me?

**Comment by javabeanz** [comment]
thanks for your report. I have the impression that this is a limitation of the Log4j library, not of the secure logging lib. Maybe we could add a unit test for detached markers. from the javadoc :: 
https://logging.apache.org/log4j/log4j-2.12.1/log4j-slf4j-impl/apidocs/org/apache/logging/slf4j/Log4jMarkerFactory.html#detachMarker-java.lang.String-

**Comment by CarlBosch** [comment]
Hi,
I've tested quickly with Log4j and got the same result. Is there a known workaround for this limitation?

**Comment by spoofzu** [comment]
As @javabeanz mentioned, no support for detached Markers in both log4j and log4j-2.  I found a workaround on old thread discussed.  Maybe it's an option for you.  

"I worked around the issue. Basically, I had a setup where I had basic
markers A, B and a marker C which existed in 2 versions:
1) C - with reference to A and 2) C with reference to B. What I did instead
was create 2 separate markers. "C_A" with references to C and A and "C_B"
with references to C and B."

[original link](http://apache-logging.6191.n7.nabble.com/Unable-to-create-detached-markers-via-log4j-slf4j-impl-2-1-td52933.html)
